packages/pigeon/lib/cpp_generator.dart

```diff
@@ -821,7 +816,7 @@ else if (const int64_t* ${pointerVariable}_64 = std::get_if<int64_t>(&args))
 \t$output = *${pointerVariable}_64;''');
           } else if (!isBuiltin) {
             indent.write(
-                'if (const flutter::EncodableMap* $pointerVariable = std::get_if<flutter::EncodableMap>(&args)) ');
+                'if (const flutter::EncodableList* $pointerVariable = std::get_if<flutter::EncodableList>(&args)) ');
```

Comment:
For future work: I believe this is also a runtime check that we shouldn't actually have (i.e., one checking for a Pigeon mistake, not a client mistake).

Reply:
https://github.com/flutter/flutter/issues/116972